crates/oxide/src/lib.rs

```diff
@@ -40,9 +44,9 @@ fn init_tracing() {
 }
 
 #[derive(Debug, Clone)]
-pub struct ChangedContent {
-    pub file: Option<PathBuf>,
-    pub content: Option<String>,
+pub enum ChangedContent<'a> {
+    File(PathBuf, Cow<'a, str>),
+    Content(String, Cow<'a, str>),
 }
```

Comment:
Updated the `ChangedContent` internally, we leaked the public implementation that was necessary to map values to the Node world (via `napi-rs`), but internally we can use a proper enum.

One reason why this is required is that we need access to the `extension`. This was only available by looking at the `file`, but we didn't have one for `content`.